src/components/common.js

```diff
@@ -555,3 +556,44 @@ export const ClipboardButton = ({ text, onClick, children, ...props }) => {
 export const HeaderRenderer = ({ name, label, sort, onSort, style, ...props }) => h(MiniSortable, { sort, field: name, onSort }, [
   div({ style: { fontWeight: 600, ...style }, ...props }, [label || Utils.normalizeLabel(name)])
 ])
+
+export const PromptedConfirmationModal = ({ title, children, confirmationPrompt = 'Delete', buttonText = 'Delete', onConfirm, onDismiss }) => {
+  const [busy, setBusy] = useState(false)
+  const [confirmation, setConfirmation] = useState('')
+
+  const del = async () => {
```

Comment:
Nit: "del" isn't a good name for this anymore. Perhaps `handleClick` or `onClick`?

Reply:
Renamed to "confirm" in [efa7c43](https://github.com/DataBiosphere/terra-ui/pull/2920/commits/efa7c4393e689704d88f2a80283f538288b314d5).